e2e/cephfs_helper.go

```diff
@@ -212,6 +212,39 @@ func listCephFSSubVolumes(f *framework.Framework, filesystem, groupname string)
 	return subVols, nil
 }
 
+type cephfsSubvolumeMetadata struct {
+	PVCNameKey      string `json:"csi.storage.k8s.io\/pvc\/name"`
+	PVCNamespaceKey string `json:"csi.storage.k8s.io\/pvc\/namespace"`
+	PVNameKey       string `json:"csi.storage.k8s.io\/pv\/name"`
+	ClusterNameKey  string `json:"csi.ceph.com\/cluster\/name"`
+}
+
+func listCephFSSubvolumeMetadata(
+	f *framework.Framework,
+	filesystem,
+	subvolume,
+	groupname string,
+) ([]cephfsSubvolumeMetadata, error) {
+	var metadata []cephfsSubvolumeMetadata
+	stdout, stdErr, err := execCommandInToolBoxPod(
+		f,
+		fmt.Sprintf("ceph fs subvolume metadata ls %s %s --group_name=%s --format=json", filesystem, subvolume, groupname),
+		rookNamespace)
+	if err != nil {
+		return metadata, err
+	}
+	if stdErr != "" {
+		return metadata, fmt.Errorf("error listing subvolume metadata %v", stdErr)
+	}
+
+	err = json.Unmarshal([]byte(stdout), &metadata)
```

Comment:
@Madhu-1 @nixpanic do you see anything wrong here, unmarshal is failing for some reason?

In the previous run, I had added a log to print stdout and this is how it looks:

```
Jul 18 14:48:08.414: INFO: ExecWithOptions: execute(POST https://192.168.39.245:8443/api/v1/namespaces/rook-ceph/pods/rook-ceph-tools-6944779c47-z9q6b/exec?command=%2Fbin%2Fsh&command=-c&command=ceph+fs+subvolume+metadata+ls+myfs+csi-vol-3ca56d9c-06a0-11ed-85d9-aedfd69586e0+--group_name%3De2e+--format%3Djson&container=rook-ceph-tools&container=rook-ceph-tools&stderr=true&stdout=true)
Jul 18 14:48:10.685: INFO: stdout=
{
    "csi.ceph.com/cluster/name": "k8s-cluster-1",
    "csi.storage.k8s.io/pv/name": "pvc-8552822b-6172-447b-9065-5a0d5954929a",
    "csi.storage.k8s.io/pvc/name": "csi-cephfs-pvc",
    "csi.storage.k8s.io/pvc/namespace": "cephfs-8081"
}
 stdErr= err=json: cannot unmarshal object into Go value of type []e2e.cephfsSubvolumeMetadata
Jul 18 14:48:10.685: FAIL: failed to list subvolume metadata: json: cannot unmarshal object into Go value of type []e2e.cephfsSubvolumeMetadata
```

Reply:
I think I got the problem